tests/lib/metadata_linux_bridge_test.py

```diff
@@ -0,0 +1,365 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import metadata
+from libnmstate import state
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import LinuxBridge
+
+
+BRIDGE0 = "br0"
+BRIDGE1 = "br1"
+IFACE0 = "eth0"
+IFACE1 = "eth1"
+
+
+class TestLinuxBridgeMetadata:
+    def test_creation_with_new_ports(self):
+        desired_state = state.State(
```

Comment:
IMHO the tests are very hard to understand since there are only subtle differences between the original and expected state that are hidden in the complex structures that create the states.

Reply:
These are unit tests, therefore there is a need to understand how metadata is behaving.
Making tests and code simpler is always welcome, so if you have a better idea, please share.

Reply:
It might be nicer to create the desired stated and the expected desired state from a common state so that it is more clear that the difference is the added metadata.

Reply:
You will just move the hard reading to the editation of the expected state. Editing an item in the list is not trivial.
In this case, it should be easy to see that the difference is with the `_create_iface_state`, where the expected is adding the metadata.